flink-runtime/src/main/scala/org/apache/flink/runtime/jobmanager/JobManager.scala

```diff
@@ -356,7 +356,12 @@ class JobManager(
         msg.resourceManager() ! decorateMessage(new TriggerRegistrationAtJobManager(self))
         // try again after some delay
         context.system.scheduler.scheduleOnce(2 seconds) {
-          self ! decorateMessage(msg)
+          currentResourceManager match {
```

Comment:
This access is problematic, because we're accessing actor state from within another thread.